network/local.go

```diff
@@ -0,0 +1,459 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/net/context"
+	"reflect"
+	"sync"
+	"time"
+)
+
+// NewLocalRouter returns a fresh router which uses local connections. It takes
+// the default context.
+func NewLocalRouter(sid *ServerIdentity) (*Router, error) {
+	return NewLocalRouterWithContext(defaultLocalContext, sid)
+}
+
+// NewLocalRouterWithContext is the same as NewLocalRouter but takes a specific
+// LocalContext. This is useful to run parallel different local overlays.
```

Comment:
> This is useful to run parallel different local overlays

In case multiple tests are to be ran in parallel, where each one listens and connect to the same set of addresses, each tests must create their own LocalContext, so each Local listeners / conn of each tests don't interfer with the other Local listeners / conn of the other tests.

Is that OK ? (I'll put that in the comment if it's OK for you, otherwise the comments won't be shown)